chatcommands.lua

```diff
@@ -298,12 +303,41 @@ if areas.factions_available then
 				return false, S("Area @1 does not exist"
 						.." or is not owned by you.", id)
 			end
-			local open = not areas.areas[id].faction_open
-			-- Save false as nil to avoid inflating the DB.
-			areas.areas[id].faction_open = open or nil
-			areas:save()
-			return true, open and S("Area opened for faction members.")
-				or S("Area closed for faction members.")
+			if factions.version == nil or factions.version < 2 or factions.mode_unique_faction then
+				local open = not areas.areas[id].faction_open and {factions.get_player_faction(name)}
+				-- Save false as nil to avoid inflating the DB.
+				areas.areas[id].faction_open = open or nil
+				areas:save()
+				return true, open and S("Area opened for faction members.")
+					or S("Area closed for faction members.")
+			else
+				local faction_name = params[2]
+				if not factions.get_owner(faction_name) then
```

Comment:
What if the faction was removed in the meantime? You could not remove the entry from the area's access list.

Reply:
In hud.lua I add a verification to remove removed factions, to display only existing faction (and it saves areas if a faction is removed) but i can add the same check here if you want

Reply:
No, I think it's fine to have it in one place only. Not sure whether HUDs are the right location but acceptable if it works like that.

Reply:
It works well ;)
Putting it here decrease the size of area.faction_open (so it decrease the number of checks) and display only existing faction.
And it saves areas only if there is some changes.